### PR TITLE
docs(labels): add category-testing label for test-related work

### LINKED ISSUE #258: Add category-testing label for test-related work
## Description

The current category label system lacks a dedicated label for test-related work. Tests are currently categorized under `category-infrastructure` (line 98 of `docs/reference/labels-and-priorities.md`), but this creates ambiguity and makes it difficult to filter and track test-specific work.

## Problem

**Current State**:
- Testing is mentioned under `category-infrastructure`: "Testing infrastructure and automation"
- Ambiguous whether this means:
  - Test setup/tooling (Jest config, test runners)
  - Test code itself (writing unit tests, fixing tests)
  - Test conventions and standards
  - All of the above?

**Real-World Impact**:
- Issue #257 had to use BOTH `category-infrastructure` AND `category-documentation` because it spans test code fixes and test convention documentation
- No clear way to filter for "all test-related issues"
- Test quality work gets mixed with DevOps/CI/CD work in issue tracking

## Proposed Solution

Add a new category label: **`category-testing` 🧪**

### Scope

Issues labeled with `category-testing` would include:

**Test Code**:
- Writing new tests (unit, integration, E2E)
- Fixing failing or flaky tests
- Improving test coverage
- Test refactoring and cleanup

**Test Infrastructure**:
- Jest/testing library configuration
- Test utilities and helpers
- Mock setup and teardown
- Test performance optimization

**Test Standards**:
- Test conventions and best practices
- Code review standards for tests
- Test documentation
- Definition of Done for testing

**Test Tooling**:
- Coverage reporting setup
- Test runners and watch mode
- Snapshot testing configuration
- Accessibility testing tools

### Updated Category Labels

```markdown
### category-testing 🧪

- **Test code** (unit, integration, E2E tests)
- **Testing infrastructure** (Jest, testing libraries, runners)
- **Test standards** and conventions
- **Test utilities** and mocking setup
- **Coverage and quality** tooling
```

## Implementation Tasks

- [ ] Create `category-testing` label in GitHub repository
- [ ] Update `docs/reference/labels-and-priorities.md` to add category-testing section
- [ ] Update examples in labels-and-priorities.md to include testing examples
- [ ] Add testing category to issue creation checklist
- [ ] Relabel existing test-related issues (starting with #257)

## Updated Category System

After this change, the four main categories would be:

1. **category-feature** 🟦 - User-facing functionality
2. **category-infrastructure** 🟧 - DevOps, CI/CD, backend architecture (excluding tests)
3. **category-testing** 🧪 - All test-related work
4. **category-documentation** 🟩 - Documentation and ADRs
5. **category-dx** 🟪 - Developer experience and tooling

## Benefits

✅ **Clarity**: Unambiguous categorization for test work
✅ **Filtering**: Easy to find all test-related issues (`label:category-testing`)
✅ **Tracking**: Better visibility into test quality efforts
✅ **Reporting**: Can measure test coverage improvement over time
✅ **Prioritization**: Test debt is visible and trackable

## Related Issues

- **#257** - Add cleanup hooks for Object.defineProperty mocks and update test conventions
  - Currently labeled: `category-infrastructure,category-documentation`
  - Should be: `category-testing` (once available)
  - This issue motivated the need for a dedicated testing category

## Example Issues for `category-testing`

- Fix flaky tests in real-time sync
- Add unit tests for new authentication feature
- Update test conventions for mock cleanup
- Improve test coverage for edge cases
- Configure snapshot testing for UI components

## Acceptance Criteria

- [ ] `category-testing` label created in GitHub
- [ ] Documentation updated in `docs/reference/labels-and-priorities.md`
- [ ] Category examples include testing scenarios
- [ ] Issue #257 relabeled with `category-testing`
- [ ] All existing test-related issues reviewed and relabeled

## Labels

- category-dx (using DX for now since this is about improving the labeling system)
- category-documentation (needs doc updates)
- priority-3-medium
- complexity-minimal

### PR DESCRIPTION
**Pull request by mikiwiik-agent**
## Summary

Introduces a dedicated `category-testing` label to clearly distinguish test-related work from infrastructure/DevOps tasks.

**Problem**: Test work was ambiguously categorized under `category-infrastructure`, making it difficult to filter and track test-specific issues.

**Solution**: New `category-testing` 🧪 label covering:
- Test code (unit, integration, E2E)
- Testing infrastructure (Jest, testing libraries)
- Test standards and conventions
- Test utilities and mocking
- Coverage and quality tooling

## Changes

- Created `category-testing` label in GitHub
- Updated `docs/reference/labels-and-priorities.md`:
  - Added category-testing section
  - Removed "Testing infrastructure" from category-infrastructure scope
  - Added testing example to Assessment Examples
- Relabeled issues #257, #196, #189 with `category-testing`

## Benefits

✅ Clear categorization for all test-related work
✅ Easy filtering with `label:category-testing`
✅ Better visibility into test quality efforts
✅ Separates test work from infrastructure/DevOps tasks

Fixes #258

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
           fetch-depth: 0  # Need full history to validate all PR commits
 
       - name: Validate PR commits
-        uses: wagoid/commitlint-github-action@v6
+        uses: wagoid/commitlint-github-action@b948419dd99f3fd78a6548d48f94e3df7f6bf3ed # v6.2.1 (Oct 19, 2025 - last known working)
```
Comment (mikiwiik) [inline .github/workflows/build.yml]: Now that we have a know workaround 
https://github.com/mikiwiik/instructions-only-claude-coding/pull/271 
for the issue we attempted to fix here, there should not be a need for this.

Please remove the fixed version workaround and validate that the build still works
